dnf/module/repo_module.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (C) 2017  Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+
+from ConfigParser import ConfigParser
+from collections import OrderedDict
+
+from dnf.conf import ModuleConf
+from dnf.exceptions import Error
+from dnf.module import module_errors, NO_STREAM_ERR, DIFFERENT_STREAM_INFO, STREAM_NOT_ENABLED_ERR
+from dnf.module.repo_module_stream import RepoModuleStream
+from dnf.util import logger, ensure_dir
+
+
+class RepoModule(OrderedDict):
+    def __init__(self):
+        super(RepoModule, self).__init__()
+
+        self._conf = None
+        self.defaults = None
+        self.name = None
+        self.parent = None
+
+    @property
+    def conf(self):
```

Comment:
Are you sure that inheritance here will work and that you will not generate new base.conf every time when you use it?